model/jpa/src/main/java/org/keycloak/models/jpa/RealmAdapter.java

```diff
@@ -793,38 +793,29 @@ public void removeDefaultRoles(String... defaultRoles) {
 
     @Override
     public Stream<GroupModel> getDefaultGroupsStream() {
-        Collection<GroupEntity> entities = realm.getDefaultGroups();
+        Collection<String> entities = realm.getDefaultGroupIds();
         if (entities == null || entities.isEmpty()) return Stream.empty();
```

Comment:
Is this check here useful/necessary? `realm.getDefaultGroupIds()` never returns null and if I understand it correctly calling `.map` on empty stream is not expensive operation. @mhajas, @hmlnarik wdyt?

Reply:
The `RealmModel.getDefaultGroupIds()` has to state that it `never returns null` to have this as a contract.

Reply:
The `getDefaultGroupIds()` here is called on the `RealmEntity` (not public interface) not on the RealmModel. Do we still need to add the `never returns null` claim?